packages/jest-haste-map/src/crawlers/node.ts

```diff
@@ -32,22 +32,42 @@ function find(
 
   function search(directory: string): void {
     activeCalls++;
-    fs.readdir(directory, (err, names) => {
+    fs.readdir(directory, {withFileTypes: true}, (err, entries) => {
       activeCalls--;
       if (err) {
         callback(result);
         return;
       }
-      names.forEach(file => {
-        file = path.join(directory, file);
-        if (ignore(file)) {
-          return;
+      entries.forEach((entry: string | fs.Dirent) => {
+        let file: string;
+        // node < v10.10 does not support the withFileTypes option, and
+        // entry will be a string.
+        if (typeof entry === 'string') {
+          file = path.join(directory, entry);
+        } else {
+          file = path.join(directory, entry.name);
```

Comment:
micro optimization, but can do this after the `ignore` and `entry.isSymobilicLink()` calls